### DISCUSSION TITLE
Custom `thumbnailTime` and `poster` with blurhash support

### DISCUSSION THREAD

**Discussion by decepulis** [muxinc/next-video#141] [Ideas]
Fleshing out the comment in #22:
1. I’d like to have a way to set `thumbnailTime` (and have the blurhash reflect that)
2. I’d like to be able to set my own custom `poster` (and have the blurhash reflect that)

If I had to implement this, I might do something like
1. Set thumbnailTime through a config file; given `videos/myVideo.mp4.json`, I could have a `videos/myVideo.thumbnailTime.txt` that could set that. It’s a weird convention, but [similar to how next does file-based metadata](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/opengraph-image#opengraph-imagealttxt), so not totally foreign.
2. Set a custom poster like that, too. `videos/myVideo.poster.jpg`
With these files present, `npx next-video sync` would calculate the new blurhash and store the thumbnailTime/poster URL in the json. The carriage files would be committed to git.

**Comment by decepulis** [comment]
Had a conversation with @luwes and he had the following thought:
The file-based API is weird. We could instead encourage users to just edit the `myVideo.mp4.json` itself with their own thumbnailTime or poster URL. If that poster is local, they could store it in the public directory and provide a relative URL.

**Comment by decepulis** [comment]
One thought that crosses my mind: how would next-video know that the thumbnailTime or posterUrl has changed, in order to recalculate the blurhash?

**Comment by heff** [comment]
fwiw, I like the general idea of being able to drop an image in the /videos folder, like `videos/myVideo.poster.jpg`. That feels very simple and intuitive. Good question on how you might update that, or any stored poster.

For a custom poster/thumbnail time, could we do something like `import myVideo from '/videos/my-video.mov?poster_time=10'`;

**Comment by luwes** [comment]
I think the prop has to be on the component because on the import URL it falls apart when using a string based source.
If the source URL is coming from a database, not static, then there is no video source import.

**Comment by heff** [comment]
For a custom poster, could we just give users something like a slot to drop in an <Image> component? That feels pretty intuitive and easy on us compared to having to manage and optimize custom images in this workflow.

**Comment by JulianJorgensen** [comment]
@heff's approach would certainly be fantastic DX for NextJS developers. The built-in <Image> component has very powerful caching etc.
